experiments/snippets/skuidCustom/skuidCustom.js

```diff
@@ -0,0 +1,1344 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-extend-native */
+/* eslint-disable no-undef */
+const $ = skuid.$;
+
+// Custom SKUID Functions
+skuid.custom = {};
+
+// Date.formatDate(date)
+// formats a javascript date as YYYY-MM-DD for use with Salesforce
+Date.formatDate = function (date) {
+	const d = new Date(date);
+	let month = `${(d.getMonth() + 1)}`;
+	let day = `${d.getDate()}`;
+	const year = d.getFullYear();
+
+	if (month.length < 2) {
+		month = `0${month}`;
+	}
+	if (day.length < 2) {
+		day = `0${day}`;
+	}
+
+	return [year, month, day].join('-');
+};
+Date.prototype.formatDate = function () { 
+	return Date.formatDate(this);
+};
+
+// Date.getDateFromSFDate(dateStr)
+// Takes a Salesforce formatted date (YYYY-MM-DD) and converts it to a Javascript Date object
+// returns Javascript date object
+Date.getDateFromSFDate = function (dateStr) {
+	if (dateStr === undefined || dateStr === null || typeof dateStr !== 'string') {
+		return undefined;
+	}
+	
+	const dateArr = dateStr.split('-');
+	
+	if (dateArr.length !== 3) {
+		return undefined;
+	}
+	
+	const year = dateArr[0];
+	const month = dateArr[1];
+	const day = dateArr[2];
+	
+	const yearInt = Number(year);
+	const monthInt = Number(month);
+	const dayInt = Number(day);
+	
+	const d = new Date(yearInt, monthInt - 1, dayInt);
+	
+	return d;
+};
+
+// Date.isLeapYear(year)
+// Checks if a year is a leap year, returns true or false
+Date.isLeapYear = function (year) {
+	return (year % 100 === 0) ? (year % 400 === 0) : (year % 4 === 0);
+};
+Date.prototype.isLeapYear = function () { 
+	return Date.isLeapYear(this.getFullYear()); 
+};
+
+// Date.getDaysInMonth(year,month)
+// Given a year and a month returns the number of days in that month. Accounts for leap years.
+Date.getDaysInMonth = function (year, month) {
+	return [31, (Date.isLeapYear(year) ? 29 : 28), 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][month];
+};
+Date.prototype.getDaysInMonth = function () { 
+	return Date.getDaysInMonth(this.getFullYear(), this.getMonth());
+};
+
+// Date.addMonths(months)
+// Adds a number of months to a date, returns the updated date
+Date.prototype.addMonths = function (months) {
+	const n = this.getDate();
+	this.setDate(1);
+	this.setMonth(this.getMonth() + months);
+	this.setDate(Math.min(n, this.getDaysInMonth()));
+	return this;
+};
+
+// Date.addDays(days)
+// adds a number of days to a date, returns the updated date
+Date.prototype.addDays = function (days) {
+	const date = new Date(this.valueOf());
+	date.setDate(date.getDate() + days);
+	return date;
+};
+
+// Object.byString(ourObject,fieldToGet);
+// Able to get object's sub-objects with a single string variable
+// instead of needing dot notation between multiple separate variables
+// eg: Object.byString(ourObject,'Field__r.SubField__c')
+// instead of needing to do ourObject['Field__r']['SubField__c'] or ourObject.Field__r.SubField__c
+Object.byString = function (ourObject, fieldToGet) {

```

Comment:
Skuid also provides a utility method for doing this: `skuid.utils.getObjectProperty(ourObject, fieldToGet)`:

https://docs.skuid.com/latest/v1/en/skuid/api/skuid_utils.html#skuid.utils.getObjectProperty

Unfortunately the docs are missing details here! I will get that sorted out.